perceval/components/core_catalog/controlled_rotation_gates.py

```diff
@@ -0,0 +1,136 @@
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+import math
+from warnings import warn
+
+from perceval.components import ACircuit, Circuit, Port, Unitary, Processor
+from perceval.components.component_catalog import CatalogItem, AsType
+from perceval.utils import Encoding, PostSelect, Matrix
+
+import numpy as np
+import cmath as cm
+from scipy.linalg import block_diag
+
+
+def build_control_gate_unitary(n: int, alpha: float) -> Matrix:
+    """ Builds the Unitary transformation of the n-qubit controlled-rotation gate C...CZ(alpha)
+
+    :param n: number of qubits of the gate
+    :param alpha: rotation angle of the gate
+    :return: Unitary matrix of the post-selected gate
+             Ref : https://arxiv.org/abs/2405.01395
+       
+    """
+
+    I = Matrix.eye(n)
+    a = (cm.exp(alpha * 1j) - 1) ** (1 / n)
+    J = np.roll(Matrix.eye(n), shift=1, axis=1)
+
+    # Main matrix block
+    A0 = I + a * J
+    B0 = Matrix.eye(n)
+    M = block_diag(B0, A0)
+
+    U0 = Matrix.get_unitary_extension(M)
+
+    # Setting the modes right for the dual rail encoding
+    initial_modes = [i for i in range(4*n)]
+    final_modes = [
+        2*i for i in range(n)] + [2*i+1 for i in range(n)] + [i for i in range(2*n, 4*n)]
+    Perm = Matrix.zeros((4*n, 4*n))
+    Perm[initial_modes, final_modes] = 1
+
+    U = Perm.T @ U0 @ Perm
+
+    return U
+
+
+class PostProcessedControledRotationsItem(CatalogItem):
+    article_ref = "https://arxiv.org/abs/2405.01395"
+    description = r"""n-qubit controlled roation gate C...CZ(alpha) with 2n heralded modes and a post-selection function"""
+    str_repr = r"""                       ╭─────╮
+ctrl0 (dual rail) ─────┤     ├───── ctrl0 (dual rail)
+                  ─────┤     ├─────
+                       │     │
+ctrl1 (dual rail) ─────┤     ├───── ctrl1 (dual rail)
+                  ─────┤     ├─────
+                          .
+                          .
```

Comment:
Can you put the 
```
.
.
.
```
between the ctrl modes and maybe replace `ctrl1` with `ctrl n` so we understand the their will be lot's of ctrl modes ?